examples/app.py

```diff
@@ -0,0 +1,179 @@
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+
+r"""Minimal Flask application example for development.
+
+
+Run ElasticSearch and RabbitMQ server and then run example development server:
+
+.. code-block:: console
+
+    $ pip install -e .[all]
+    $ cd examples
+    $ ./app-recreate.sh
+
+Try to get some records:
+
+.. code-block:: console
+
+    $ curl -X GET localhost:9200/_cat/indices?v
+    $ curl -X GET localhost:9200/testrecords-testrecord-v1.0.0/_search | \
+        python -m json.tool
+"""
+
+from __future__ import absolute_import, print_function
+
+import os
+
+from flask import Flask
+from flask_celeryext import FlaskCeleryExt
+from flask_cli import FlaskCLI
+from invenio_db import InvenioDB, db
+from invenio_pidstore import InvenioPIDStore
+from invenio_records import InvenioRecords
+from invenio_search import InvenioSearch
+
+from invenio_indexer import InvenioIndexer
+
+# create application's instance directory. Needed for this example only.
+current_dir = os.path.dirname(os.path.realpath(__file__))
+instance_dir = os.path.join(current_dir, 'app_instance')
+if not os.path.exists(instance_dir):
+    os.makedirs(instance_dir)
+
+# Create Flask application
+index_name = 'testrecords-testrecord-v1.0.0'
+app = Flask(__name__, instance_path=instance_dir)
+app.config.update(
+    CELERY_ALWAYS_EAGER=True,
+    CELERY_CACHE_BACKEND='memory',
+    CELERY_EAGER_PROPAGATES_EXCEPTIONS=True,
+    CELERY_RESULT_BACKEND='cache',
+    SQLALCHEMY_TRACK_MODIFICATIONS=True,
+    INDEXER_DEFAULT_INDEX=index_name,
+    INDEXER_DEFAULT_DOC_TYPE='testrecord-v1.0.0',
+    SQLALCHEMY_DATABASE_URI=os.getenv('SQLALCHEMY_DATABASE_URI',
+                                      'sqlite:///app.db'),
+)
+
+FlaskCLI(app)
+FlaskCeleryExt(app)
+InvenioDB(app)
+InvenioPIDStore(app)
+InvenioRecords(app)
+search = InvenioSearch(app)
+search.register_mappings('testrecords', 'data')
+InvenioIndexer(app)
+
+# A few documents which will be added in order to make search interesting
+record_examples = [{
+    'title': 'Awesome meeting report',
+    'description': 'Notes of the last meeting.',
+    'participants': 42,
+    'type': 'report',
+}, {
+    'title': 'Furniture order',
+    'description': 'Tables for the meeting room.',
+    'type': 'order',
+}]
+record_examples += [{
+    'title': 'LHC experiment {}'.format(idx),
+    'description': 'Data from experiment {}.'.format(idx),
+    'type': 'data',
+} for idx in range(20)]
+
+
+@app.cli.group()
+def fixtures():
+    """Command for working with test data."""
+
+
+@fixtures.command()
+def records():
+    """Load test data fixture."""
+    import uuid
+    from invenio_records.api import Record
+    from invenio_pidstore.models import PersistentIdentifier, PIDStatus
+
+    # Record 1 - Live record
+    with db.session.begin_nested():
```

Comment:
Do we need the same copy&paste example as in invenio-records-rest? IMHO here the PIDs are not important. It should be enough to use only invenio-records.